src/gt4py/backend/gt_backends.py

```diff
@@ -656,15 +656,19 @@ def generate_extension(self, **kwargs: Any) -> Tuple[str, str]:
 
 class GTCUDAPyModuleGenerator(gt_backend.CUDAPyExtModuleGenerator):
     def generate_pre_run(self) -> str:
-        field_names = self.args_data["field_info"].keys()
+        field_names = [
```

Comment:
Given the modifications made in `base.py`, the new checks introduced here have no effect. But may be better to keep them for the future.

Reply:
Since I think the modifications in `base.py` involve a change too large for this issue, I think it's better to only keep this change in the PR.